src/v/s3/client.cc

```diff
@@ -144,7 +144,6 @@ result<http::client::request_header> request_creator::make_get_object_request(
     header.insert(boost::beast::http::field::host, host);
```

Comment:
> The signature key is valid for two days so the implementation is
caching it for performance reasons.
This solution led to some bugs when persistent connections are
involved. The bugs are particulariliy dangerous since the log uploads
are not possible until the signature key is updated.
The botocore library doesn't cache the signature key. We can assume
that it's perfectly fine for us to not to cache it as well. This commit
removes this logic.

When describing a solution to a bug in a commit message, please try to also explain what the bug is so that the commit is self contained.

Reply:
fixed that